pkg/expr/mathexp/type_series.go

```diff
@@ -48,8 +48,29 @@ func SeriesFromFrame(frame *data.Frame) (s Series, err error) {
 			s.ValueIdx = i
 		default:
 			// Handle default case
+			// try to convert to *float64
+			var convertedField *data.Field
+			for j := 0; j < field.Len(); j++ {
+				ff, err := field.FloatAt(j)
+				if err != nil {
+					break
+				}
+				if convertedField == nil { // initialise field
+					convertedField = data.NewFieldFromFieldType(data.FieldTypeNullableFloat64, field.Len())
```

Comment:
I'm probably out of context here, but this seems to me we're converting everything to nullable float64 rather than accepting any scalar value. But I'm probably confused

Reply:
Indeed, this is the default case (if the field is not one of: `data.FieldTypeTime`, `data.FieldTypeNullableTime`, `data.FieldTypeFloat64`, `data.FieldTypeNullableFloat64`)
and [field.FloatAt](https://pkg.go.dev/github.com/grafana/grafana-plugin-sdk-go@v0.103.0/data#Field.FloatAt) in line 52 will try to convert every other type otherwise - if not convertible - it will return an error.
But maybe it's true that we want only one value field; why to convert everything else 🤔  ?

Reply:
SSE currently is designed to handle multi-valued series (that is, it expects one series / number per label within a query, not multiple series or numbers per label within query).